app/controllers/user_filters_controller.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+# UserFiltersController: used to manage user saved advanced filters
+class UserFiltersController < ApplicationController
+  before_action :authenticate_user!
+  before_action :check_role
+
+  def index
+    render json: current_user.user_filters
+  end
+
+  def create
+    redirect_to root_url unless current_user.user_filters.count < 100 # Enforce upper limit per user
+    active = params.require(:active).collect do |filter|
+      {
+        filterOption: filter.require(:filterOption).permit(:name, :title, :description, :type, options: []),
+        value: filter.require(:value),
+        dateOption: filter.permit(:dateOption)[:dateOption]
+      }
+    end
+    name = params.require(:name)
+    render json: UserFilter.create!(contents: active, name: name, user_id: current_user.id)
```

Comment:
I actually want it to throw an exception, since this should never happen (and if it does, I want to see it in Sentry). That distinction (not using bang methods in prod) is new to me. Does that come from the rails docs? This is also ActiveRecord specific.